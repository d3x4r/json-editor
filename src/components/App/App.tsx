import React, { useState, useEffect } from 'react';
import PageHeader from '../Header';
import DataForm from '../DataForm';
import PreviewForm from '../PreviewForm';
import Result from '../Result';
import { Layout } from 'antd';
import { Row, Col } from 'antd';

const { Header, Content } = Layout;

const App: React.FC = () => {
  const [dataToRender, updateDataToRender] = useState({});

  const [editorState, updateEditorState] = useState(dataToRender);

  useEffect(() => {
    updateEditorState((state: {}) => {
      return dataToRender;
    });
  }, [dataToRender]);
  return (
    <Layout className="layout">
      <Header>
        <PageHeader />
      </Header>
      <Content style={{ padding: '25px 75px' }}>
        <Row gutter={[8, 8]}>
          <Col span={5}>
            <DataForm updateEditorData={updateDataToRender} />
          </Col>
          <Col span={14}>
            <PreviewForm
              data={editorState}
              calculateResult={updateDataToRender}
              updateEditorState={updateEditorState}
              dataToRender={dataToRender}
            />
          </Col>
          <Col span={5}>
            <Result data={dataToRender} />
          </Col>
        </Row>
      </Content>
    </Layout>
  );
};

export default App;
