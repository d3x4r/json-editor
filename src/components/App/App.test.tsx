import React from 'react';
import { render } from '@testing-library/react';
import App from './App';

it('Test app render', () => {
  const result = render(<App />);
  expect(result).toMatchSnapshot();
});
