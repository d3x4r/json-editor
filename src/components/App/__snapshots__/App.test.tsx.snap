// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test app render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="ant-layout layout"
      >
        <header
          class="ant-layout-header"
        >
          <div
            class="ant-page-header header ant-page-header-ghost"
          >
            <div
              class="ant-page-header-heading"
            >
              <div
                class="ant-page-header-heading-left"
              >
                <span
                  class="ant-page-header-heading-title"
                  title="JSON editor"
                >
                  JSON editor
                </span>
              </div>
            </div>
          </div>
        </header>
        <main
          class="ant-layout-content"
          style="padding: 25px 75px;"
        >
          <div
            class="ant-row"
            style="margin: -4px -4px 4px -4px;"
          >
            <div
              class="ant-col ant-col-4"
              style="padding: 4px 4px 4px 4px;"
            >
              <section
                class="data-form"
              >
                <h2
                  class="data-form__title"
                >
                  Enter data
                </h2>
                <form
                  class="ant-form ant-form-horizontal"
                  data-testid="data-form"
                >
                  <div
                    class="ant-row ant-form-item"
                  >
                    <div
                      class="ant-col ant-form-item-control"
                    >
                      <div
                        class="ant-form-item-control-input"
                      >
                        <div
                          class="ant-form-item-control-input-content"
                        >
                          <textarea
                            class="ant-input"
                            data-testid="data-form-textarea"
                            placeholder="enter data in json format"
                            rows="12"
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="ant-row ant-form-item"
                  >
                    <div
                      class="ant-col ant-form-item-control"
                    >
                      <div
                        class="ant-form-item-control-input"
                      >
                        <div
                          class="ant-form-item-control-input-content"
                        >
                          <button
                            class="ant-btn ant-btn-primary"
                            data-testid="form-data-submit"
                            disabled=""
                            type="submit"
                          >
                            <span>
                              Render Preview
                            </span>
                          </button>
                          <button
                            class="ant-btn ant-btn-default"
                            data-testid="form-data-example"
                            type="button"
                          >
                            <span>
                              Paste Example
                            </span>
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                </form>
              </section>
            </div>
            <div
              class="ant-col ant-col-14"
              style="padding: 4px 4px 4px 4px;"
            >
              <section
                class="preview-form"
              >
                <h2
                  class="preview-form__title"
                >
                  Preview
                </h2>
                <div
                  class="preview-form__result"
                >
                  <p
                    class="preview-form__message"
                  >
                    No data to render
                  </p>
                </div>
              </section>
            </div>
            <div
              class="ant-col ant-col-6"
              style="padding: 4px 4px 4px 4px;"
            >
              <section
                class="preview-form"
              >
                <div
                  class="preview-form__header"
                >
                  <h2
                    class="preview-form__title"
                  >
                    Result
                  </h2>
                  <div>
                    <span>
                      Minify: 
                    </span>
                    <button
                      aria-checked="false"
                      class="ant-switch"
                      role="switch"
                      type="button"
                    >
                      <span
                        class="ant-switch-inner"
                      />
                    </button>
                  </div>
                </div>
                <div
                  class="result-container"
                  data-testid="result-container"
                >
                  <pre
                    class="result-container__text"
                  >
                    {}
                  </pre>
                </div>
              </section>
            </div>
          </div>
        </main>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="ant-layout layout"
    >
      <header
        class="ant-layout-header"
      >
        <div
          class="ant-page-header header ant-page-header-ghost"
        >
          <div
            class="ant-page-header-heading"
          >
            <div
              class="ant-page-header-heading-left"
            >
              <span
                class="ant-page-header-heading-title"
                title="JSON editor"
              >
                JSON editor
              </span>
            </div>
          </div>
        </div>
      </header>
      <main
        class="ant-layout-content"
        style="padding: 25px 75px;"
      >
        <div
          class="ant-row"
          style="margin: -4px -4px 4px -4px;"
        >
          <div
            class="ant-col ant-col-4"
            style="padding: 4px 4px 4px 4px;"
          >
            <section
              class="data-form"
            >
              <h2
                class="data-form__title"
              >
                Enter data
              </h2>
              <form
                class="ant-form ant-form-horizontal"
                data-testid="data-form"
              >
                <div
                  class="ant-row ant-form-item"
                >
                  <div
                    class="ant-col ant-form-item-control"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <textarea
                          class="ant-input"
                          data-testid="data-form-textarea"
                          placeholder="enter data in json format"
                          rows="12"
                        />
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-row ant-form-item"
                >
                  <div
                    class="ant-col ant-form-item-control"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <button
                          class="ant-btn ant-btn-primary"
                          data-testid="form-data-submit"
                          disabled=""
                          type="submit"
                        >
                          <span>
                            Render Preview
                          </span>
                        </button>
                        <button
                          class="ant-btn ant-btn-default"
                          data-testid="form-data-example"
                          type="button"
                        >
                          <span>
                            Paste Example
                          </span>
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </form>
            </section>
          </div>
          <div
            class="ant-col ant-col-14"
            style="padding: 4px 4px 4px 4px;"
          >
            <section
              class="preview-form"
            >
              <h2
                class="preview-form__title"
              >
                Preview
              </h2>
              <div
                class="preview-form__result"
              >
                <p
                  class="preview-form__message"
                >
                  No data to render
                </p>
              </div>
            </section>
          </div>
          <div
            class="ant-col ant-col-6"
            style="padding: 4px 4px 4px 4px;"
          >
            <section
              class="preview-form"
            >
              <div
                class="preview-form__header"
              >
                <h2
                  class="preview-form__title"
                >
                  Result
                </h2>
                <div>
                  <span>
                    Minify: 
                  </span>
                  <button
                    aria-checked="false"
                    class="ant-switch"
                    role="switch"
                    type="button"
                  >
                    <span
                      class="ant-switch-inner"
                    />
                  </button>
                </div>
              </div>
              <div
                class="result-container"
                data-testid="result-container"
              >
                <pre
                  class="result-container__text"
                >
                  {}
                </pre>
              </div>
            </section>
          </div>
        </div>
      </main>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
